R/transform-calculate.R

```diff
@@ -86,23 +86,33 @@ collateCats <- function (inserts, var_cats) {
     # setup an empty AbstractCategories object to collate into
     cats_out <- AbstractCategories()
     cats_out@.Data <- var_cats
+    
+    last_category <- tail(ids(var_cats), 1)
 
     # for each insert, find the position for its anchor, and add the insertion
     # at that position we use a for loop, because as we insert, the positions of
-    # categories (which may serve as anchors) will change.
-    for (insert in inserts) {
-        pos <- findInsertPosition(insert, cats_out)
+    # categories (which may serve as anchors) will change. We also reverse the
+    # list because for insertions that have the same anchor, we want to maintain
+    # the order as it is stored in the API. To do this we insert the last one 
+    # first so that when we insert ones before that they are higher up (closer 
+    # to the category anchor)
+    # Categoies: A, B, C 
+    # Insertions: [{name = first, anchor = A}, {name = second, anchor = A}]
+    # Desired result: A, first, second, B, C
+    # Result if not `rev(inserts)`: A, second, first, B, C
+    for (insert in rev(inserts)) {
+        pos <- findInsertPosition(insert, cats_out, last_category)
```

Comment:
Does this fail if `last_category` is `NULL`? If so are we sure that `ids(var_cats)` has entries?

Reply:
Good catch, it generally doesn't fail, unless the insertion was being inserted at the end. I've added a check to `collateCats` to confirm that there is at least 1 category before proceeding which should prevent this as a possibility (since all categories have to have an id).